tokio/src/task/mod.rs

```diff
@@ -363,49 +381,74 @@ cfg_rt_core! {
         pub(crate) fn header(&self) -> &Header {
             self.raw.header()
         }
+    }
 
-        pub(crate) fn into_raw(self) -> NonNull<Header> {
-            let raw = self.raw.into_raw();
-            mem::forget(self);
-            raw
+    impl<S: 'static> Notified<S> {
+        pub(crate) unsafe fn from_raw(ptr: NonNull<Header>) -> Notified<S> {
+            Notified(Task::from_raw(ptr))
+        }
+
+        pub(crate) fn header(&self) -> &Header {
+            self.0.header()
         }
     }
 
-    impl<S: Schedule> Task<S> {
-        /// Returns `self` when the task needs to be immediately re-scheduled
-        pub(crate) fn run<F>(self, mut executor: F) -> Option<Self>
-        where
-            F: FnMut() -> Option<NonNull<S>>,
-        {
-            if unsafe {
-                self.raw
-                    .poll(&mut || executor().map(|ptr| ptr.cast::<()>()))
-            } {
-                Some(self)
-            } else {
-                // Cleaning up the `Task` instance is done from within the poll
-                // function.
+    cfg_rt_threaded! {
+        impl<S: 'static> Task<S> {
```

Comment:
It's weird to me that these next few `impl`s are under `cfg_rt_threaded`.

Reply:
Could you elaborate? Those impls are only needed for the threaded scheduler. It does imply that it could be cleaned up some... but I am opting to punt that.

Reply:
I guess we'd prefer this over `#[allow(dead_code)]`.